packages/subproviders/src/subproviders/pk_wallet_subprovider.ts

```diff
@@ -0,0 +1,68 @@
+import { assert } from '@0xproject/assert';
+import { JSONRPCRequestPayload } from '@0xproject/types';
+import EthereumTx = require('ethereumjs-tx');
+import * as ethUtil from 'ethereumjs-util';
+import * as _ from 'lodash';
+
+import { Callback, ErrorCallback, PartialTxParams, ResponseWithTxParams, WalletSubproviderErrors } from '../types';
+
+import { BaseWalletSubprovider } from './base_wallet_subprovider';
+import { Subprovider } from './subprovider';
+
+/**
+ * This class implements the [web3-provider-engine](https://github.com/MetaMask/provider-engine) subprovider interface.
+ * This subprovider intercepts all account related RPC requests (e.g message/transaction signing, etc...)
+ */
+export class PKWalletSubprovider extends BaseWalletSubprovider {
```

Comment:
I'd prefer we don't abbreviate this: `PrivateKeyWalletSubprovider`. We don't use many abbreviations and I think it lifts a cognitive burden from readers.